arc/alkali_atom_functions.py

```diff
@@ -841,9 +854,45 @@ def getQuantumDefect(self, n, l, j, s=0.5):
                 + modifiedRRcoef[5] / ((n - modifiedRRcoef[0]) ** 10)
             )
         else:
-            # use \delta_\ell = \delta_g * (4/\ell)**5
+            n = int(n)
+            l = int(l)
+    
+
+            # Use \delta_\ell = \delta_g * (4/\ell)**5
             # from https://journals.aps.org/pra/abstract/10.1103/PhysRevA.74.062712
-            defect = self.quantumDefect[0][4][0] * (4 / l) ** 5
+    
+            # Calculate r4 and r6 coefficients
+            top_r4 = 3 * pow(n, 2) - (l * (l + 1))
+            bottom_r4 = (
+                2 * pow(n, 5) * (l + 1.5) * (l + 1) * (l + 0.5) * l * (l - 0.5)
+            )
+            r4 = top_r4 / bottom_r4
+    
+            top_r6 = (
+                35 * pow(n, 4)
+                - 5 * pow(n, 2) * (6 * l * (l + 1) - 5)
+                + 3 * (l + 2) * (l + 1) * l * (l - 1)
+            )
+            bottom_r6 = (
+                8
+                * pow(n, 7)
+                * (l + 2.5)
+                * (l + 2)
+                * (l + 1.5)
+                * (l + 1)
+                * (l + 0.5)
+                * l
+                * (l - 0.5)
+                * (l - 1)
+                * (l - 1.5)
+            )
+            r6 = top_r6 / bottom_r6
+    
+            # Calculate the energy difference from polarisation energy
+            defect = (
+                0.5 * (self.a_d_eff * r4 + self.a_q_eff * r6)
+            )
+
```

Comment:
I believe above is correct based on your numerical results, however, is there some reference where these equations have been originally derived, that we can reference in commend directly in code?

Reply:
I believe the original is from J. E. Mayer and M. G. Mayer, Phys. Bev. 43, 605
(1933). 

The Freeman and Kleppner Physical Review A 14.5 (1976): 1614 I reference cites this work.

On overview is also given in pgs 340-364 `Spectroscopy_of_alkali_Rydberg_states' in Gallagher's Rydberg atoms (which they themselves cite Mayer and Mayer as the origin)

Reply:
if you are referring to the expectation values see https://journals.aps.org/pra/abstract/10.1103/PhysRevA.42.1123 or more appropriately https://journals.aps.org/pra/abstract/10.1103/PhysRevA.9.1087